osh/builtin_trap.py

```diff
@@ -95,12 +64,101 @@ def _GetSignalNumber(sig_spec):
 # Then hit Ctrl-C.
 
 
+class TrapState(pyos.SignalHandler):
```

Comment:
I don't think the TrapState should inherit from pyos.SignalHandler

To clarify the code I might name it `SharedTrapState`  -- it should be ONLY the stuff that's shared between main thread and signal handler

It's the stuff we want to test for races.  I posted some messages on `#blog-ideas` about "functions and data" and ADTs that might help illuminate how I'm thinking about it

(And also evidence from CPython, bash, dash on `#oil-dev` )